src/mainwindow.cpp

```diff
@@ -728,11 +718,14 @@ void MainWindow::openWarpMap(QString map_name, QString warp_num) {
         return;
     }
 
-    QList<Event*> warp_events = editor->map->events["warp_event_group"];
-    if (warp_events.length() > warpNum) {
-        Event *warp_event = warp_events.at(warpNum);
+    // Select the target event.
+    if (event_group != EventGroup::Warp && event_index)
```

Comment:
Can you add a comment here explaining why non-warp events use a 1-based `event_index`?